tiledb/sm/serialization/array_schema.cc

```diff
@@ -127,11 +127,12 @@ Status filter_pipeline_from_capnp(
   for (auto filter_reader : filter_list_reader) {
     FilterType type = FilterType::FILTER_NONE;
     RETURN_NOT_OK(filter_type_enum(filter_reader.getType().cStr(), &type));
-    tdb_unique_ptr<Filter> filter(Filter::create(type));
-    if (filter == nullptr)
+    auto&& [st_filter, filter_ptr]{tiledb::sm::Filter::create(type)};
```

Comment:
It's not necessary to change the capnp serialization in this PR. It will eventually need to use C.41-compliant constructors, but this can be done later.